### PR TITLE
recreate serverless sid if not found

### PR DESCRIPTION
**Pull request by ktalebian**
* Recreate serverless instance if deleted

### CODE REVIEW

File: src/__tests__/clients/FlexConfigurationClient.test.ts [hunk 1]
```diff
@@ -39,4 +60,77 @@ describe('FlexConfigurationClient', () => {
...
+      const updateServerlessSids = sinon.stub(client, 'updateServerlessSids');
+      updateServerlessSids.returnsThis();
+      await client.registerServerlessSid('ZS123');
+
+      expect(fetch).to.have.been.calledOnce;
+      expect(updateServerlessSids).not.to.have.been.called;
+      updateServerlessSids.restore();
+    });
+
+    it('should do nothing if sid is already present', async () => {
```
Comment (ahcai) [inline src/__tests__/clients/FlexConfigurationClient.test.ts:82]: wrong test description

File: src/clients/FlexConfigurationClient.ts [hunk 2]
```diff
@@ -1,31 +1,103 @@
...
+    return this.fetch();
+  }
+
+  /**
+   * Removes a Serverless sid
+   * @param serviceSid the sid to remove
+   */
+  public async unregisterServerlessSid(serviceSid: string): Promise<ConfigurationInstance> {
+    const config = await this.fetch();
+    const index = config.serverlessServiceSids.indexOf(serviceSid);
```
Comment (ahcai) [inline src/clients/FlexConfigurationClient.ts:68]:
```
if (!config.serverlessServiceSids.includes(serviceSid)) {
  return config;
}
```

File: src/clients/FlexConfigurationClient.ts [hunk 3]
```diff
@@ -1,31 +1,103 @@
...
+   * @param sids  the serverless sid to update
+   * @private
+   */
+  private async updateServerlessSids(sids: string[]): Promise<void> {
+    const auth = Buffer.from(`${this.auth.username}:${this.auth.password}`, 'utf8').toString('base64');
+    // eslint-disable-next-line camelcase
+    const data = { account_sid: this.auth.accountSid, serverless_service_sids: sids };
+
+    const response = await phin({
+      url: 'https://flex-api.twilio.com/v1/Configuration',
```
Comment (ahcai) [inline src/clients/FlexConfigurationClient.ts:90]: can we make this url realm-friendly?

File: src/clients/FlexConfigurationClient.ts (hunk 2 above)
Comment (ktalebian) [reply src/clients/FlexConfigurationClient.ts:68]: I do need the index if it does exist to remove; hence why I'm using this.

Comment (ahcai) [reply src/clients/FlexConfigurationClient.ts:69]: oops didn't notice you were using it below 👍

**Review by ahcai — APPROVED** [review]

**Pull request merged**